### PR TITLE
Plugin Dependencies

### PR DESCRIPTION
**Pull request by alMukaafih**

### CODE REVIEW

File: src/lib/installPlugin.js [hunk 1]
```diff
@@ -139,3 +155,99 @@ async function createFileRecursive(parent, dir) {
...
+		)
+			.readFile("json")
+			.catch(() => null);
+
+		if (!remoteDependency) return true;
+
+		const version = await isInstalled(id);
+		if (remoteDependency?.version === version) return false;
+
+		plugin = Object.assign({}, remoteDependency);
```
Comment (bajrangCoder) [inline src/lib/installPlugin.js]: i guess you forget to define this variable

**Comment by bajrangCoder** [comment]
have you tested it? It doesn't work

**Comment by alMukaafih** [comment]
I have tested it and it works for  both free and paid plugins

File: src/lib/installPlugin.js (hunk 1 above)
Comment (alMukaafih) [reply src/lib/installPlugin.js]: The variable is unused

**Comment by bajrangCoder** [comment]
> I have tested it and it works for both free and paid plugins though further testing is needed for paid plugins

Now it's working fine.
Just one thing left:
It should show the name of plugin which is installing in loader and also before installing the dependencies it's nice to show the user that it's going to install these plugins too.

**Comment by alMukaafih** [comment]
> > I have tested it and it works for both free and paid plugins though further testing is needed for paid plugins
> 
> Now it's working fine. Just one thing left: It should show the name of plugin which is installing in loader and also before installing the dependencies it's nice to show the user that it's going to install these plugins too.

Sorry for the late response, was facing some issues irl.

I am deliberately either to show an `alert` for the dependencies that are going to be installed or a `confirmation`.
Something like
```js
alert(
    "Installer Notice",
    "Acode is going to install the following dependencies<br/>"
      + plugin.dependencies.join("<br/>"),
    () => {
        // Resolve dependencies
    },
);
```
Or
```js
const shouldInstall = await confirm(
    "Installer Notice",
    "Acode is going to install the following dependencies<br/>"
      + plugin.dependencies.join("<br/>"),
);

if (shouldInstall) {
    // Resolve dependencies 
}

**Comment by bajrangCoder** [comment]
Use confirmation

**Comment by alMukaafih** [comment]
> Use confirmation

Ok 👍

**Review by bajrangCoder — APPROVED** [review]

**Review by Reybabylon — APPROVED** [review]

**Review by Reybabylon — APPROVED** [review]

**Pull request merged**